### PR TITLE
DR 2763 backfill firstindexed dt

### PR DESCRIPTION
**Pull request by croyfish**
These are the code changes used to update timestamps in repoapi qa. When we're ready to update rsolr in production, this can be brought into the development pipeline.

### CODE REVIEW

File: app/jobs/dt_update_job.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      update_params = { params: { commit: true }, data: update_data.to_json }
+      response = solr.update(update_params)
+    rescue StandardError => e
+      puts "Got a hiccup. Going one by one."
+      update_data.each do |datum|
+        update_params = { params: { commit: true }, data: [datum].to_json }
+        begin
+          response = solr.update(update_params)
+        rescue StandardError => e
+          datum[:firstIndexed_dt] = { set: DateTime.parse(datum[:firstIndexed_dt][:set].to_s).utc.strftime('%Y-%m-%dT%H:%M:%SZ') }
```
Comment (croyfish) [inline app/jobs/dt_update_job.rb:45]: Probably the conditional dt transformation can be brought into the main loop by using a regex to check the existing format, but this hasn't yet been proven out in terms of performance.

Execution against QA took 8 hours for the successful "main" jobs and another 10 hours to process the failed jobs that then had to go one-by-one.

It's also questionable whether we can direct this much load against production solr for this long. The processing may need to be bached into several off-hours windows to complete without negatively impacting patron-facing production systems.

File: scripts/backfill_firstindexed_dt.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+require 'rsolr'
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'config', 'environment'))
+
+ActiveRecord::Base.logger.level = 2
+
+start_time = Time.now
+
+# Solr URL
+solr_url = 'http://10.225.132.218:8983/solr/repoapi'
```
Comment (croyfish) [inline scripts/backfill_firstindexed_dt.rb:11]: This'll need to be updated for prod

File: app/jobs/dt_update_job.rb (hunk 1 above)
Comment (croyfish) [reply app/jobs/dt_update_job.rb:45]: Notes: Also consider reducing the batch size and enqueueing the jobs at a lower priority if they end up getting enqueued in the prod db

File: scripts/backfill_firstindexed_dt.rb (hunk 2 above)
Comment (emu47) [reply scripts/backfill_firstindexed_dt.rb:11]: Could this be an argument passed to the script instead of hard-coded?

**Pull request closed without merging**